lib/utils/errorHandler.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+var chalk = require('chalk');
+
+var errors = [];
+var unsavedErrors = {};
```

Comment:
I still don't quite get the difference between errors and unsavedErrors

Reply:
From the grouping POV, unsavedErrors are still being grouped and errors have already been grouped.

There is a bit more to it than that, e.g. handling errors that don't need grouping but you want to wait to output. But its just a generalization of the above case where group size is 1.